fda-model/hydraulics/HydraulicDataset.cs

```diff
@@ -15,6 +20,34 @@ public class HydraulicDataset
 
         public List<IHydraulicProfile> HydraulicProfiles { get; } = new List<IHydraulicProfile>();
         public HydraulicDataSource DataSource { get; set; }
+        public List<UncertainPairedData> GetGraphicalStageFrequency(string pointShapefileFilePath, string parentDirectory)
+        {
+            List<UncertainPairedData> ret = new List<UncertainPairedData>();
+            PointFeatureLayer indexPoint = new PointFeatureLayer("ThisNameIsNotUsedForAnythingHere", pointShapefileFilePath);
+            PointMs indexPoints = new PointMs(indexPoint.Points().Select(p => p.PointM()));
+            for (int j = 0; j < indexPoints.Count; j++)
+            {
+                double[] probs = new double[HydraulicProfiles.Count];
+                float[] wses = new float[HydraulicProfiles.Count];
+                for (int i = 0; i < HydraulicProfiles.Count; i++)
+                {
+                    PointM pt = indexPoints[j];
+                    PointMs converted = new PointMs();
+                    converted.Add(pt);
+                    float[] singleWSE = HydraulicProfiles[i].GetWSE(converted, DataSource, parentDirectory);
+                    probs[i] = HydraulicProfiles[i].Probability;
+                    wses[i] = singleWSE[0];
+                }
+                IDistribution[] distributions = new IDistribution[wses.Length];
+                for( int i=0; i<wses.Length; i++)
+                {
+                    distributions[i] = new Deterministic(wses[i]);
+                }
+                UncertainPairedData pd = new(probs, distributions, new CurveMetaData());
```

Comment:
will we need curve metadata?

Reply:
I'd like to test on your machine with Muncie. Strange that you're not getting curves. I may need to add some messaging.

Reply:
I don't think so. Though we didn't design it well to know when we need it. I'm hoping to leave it blank until It comes up that i need something from it. Nothing in this design demands it yet.